src/main/java/com/example/CourseWork2/service/JavaQuestionService.java

```diff
@@ -0,0 +1,58 @@
+package com.example.CourseWork2.service;
+
+import com.example.CourseWork2.entity.Question;
+import com.example.CourseWork2.excrption.QuestionAlreadyAddedException;
+import com.example.CourseWork2.excrption.QuestionNotFoundException;
+import org.springframework.context.annotation.AnnotationConfigApplicationContextExtensionsKt;
+import org.springframework.stereotype.Service;
+import java.util.Random;
+
+import java.util.*;
+
+@Service
+public class JavaQuestionService implements QuestionService{
+    Set<Question> questions = new HashSet<>();
+
+    public Question add(String question, String answer) {
+        Question newQuestion = new Question(question, answer);
+        if (questions.contains(newQuestion)){
+            throw new QuestionAlreadyAddedException("Такой вопрос уже добавлен в список");
+        }
+        questions.add(newQuestion);
+        return newQuestion;
+    }
+
+    @Override
+    public Question add(Question question) {
+        for (Question q:questions) {
+            if (q.equals(question)){
+                throw new QuestionAlreadyAddedException("Такой вопрос уже добавлен в список");
+            }
+        }
+        questions.add(question);
+        return question;
```

Comment:
просто
```suggestion
        if (!questions.add(question)) {
            throw new QuestionAlreadyAddedException("Такой вопрос уже добавлен в список");
        }
        return question;
```
надо внимательнее изучать и читать методы коллекций, что они возвращают - это поможет писать код лаконичнее и короче